basic_subscriber.go

```diff
@@ -79,10 +81,43 @@ func (s *BasicSubscriber) Subscribe(ctx context.Context, gotData func(seqNo uint
 		glog.V(5).Infof("New peer from kademlia: %v", peer.IDHexEncode(p))
 		ns := s.Network.NetworkNode.GetOutStream(p)
 		if ns != nil {
-			//Send SubReq
-			glog.Infof("Sending Req %v", s.StrmID)
-			if err := s.Network.sendMessageWithRetry(p, ns, SubReqID, SubReqMsg{StrmID: s.StrmID}); err != nil {
-				glog.Errorf("Error sending SubReq to %v: %v", peer.IDHexEncode(p), err)
+			if targetPid == p {
+				// Send SubReq
+				glog.Infof("Already have a direct cxn to broadcaster; subscribing")
+				sub := SubReqMsg{StrmID: s.StrmID}
+				err = s.Network.sendMessageWithRetry(p, ns, SubReqID, sub)
+				if err != nil {
+					glog.Errorf("Error sending SubReq to broadcaster : %v", err)
+					return err
+				}
+			} else {
+				// Send TranscodeSub
+				glog.Infof("%v Sending TranscodeSub to %v", s.Network.NetworkNode.ID(), p)
+				localNodeID := peer.IDB58Encode(s.Network.NetworkNode.ID())
+				ipfs, err := ma.NewMultiaddr("/ipfs/" + localNodeID)
+				if err != nil {
+					glog.Errorf("Unable to create IPFS multiaddr for local node : %v", err)
+					return err
+				}
+				maddrs := make([]ma.Multiaddr, len(s.Network.NetworkNode.Host().Addrs()))
+				for i, v := range s.Network.NetworkNode.Host().Addrs() {
```

Comment:
I haven't used encapsulation before.  Is there a reason why we use the IPFS encapsulation instead of sending the nodeID and multiaddr both in `TranscodeSub`?

Reply:
No reason other than an overly literal reading of what we discussed in https://github.com/livepeer/go-livepeer-basicnet/issues/21. Fixed to add a `NodeID` to the struct rather than appending it to each multiaddr in the list. Good call.

Reply:
Fixed in https://github.com/livepeer/go-livepeer-basicnet/pull/32/commits/f422c89b3d24dffa5ba132b5f3c323ff412fe89b